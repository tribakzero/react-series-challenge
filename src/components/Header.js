import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import SearchBar from './SearchBar'

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchValue: ''
    }
  }
  handleSearch = (e) => {
    console.log(e.target.value)
  }
  render() {
    return (
      <header className="App-header">
        <NavLink to="/" activeClass="active" exact>Home</NavLink>
        <NavLink to="/favorites" activeClass="active">Favorites</NavLink>
        <SearchBar value={this.searchValue} onChange={this.handleSearch} />
      </header>
    )  
  }
}

export default Header;
