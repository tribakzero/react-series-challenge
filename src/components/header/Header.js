import React from 'react';
import { NavLink } from 'react-router-dom';
import { StyledHeader } from './Header.style'

const Header = () => {
  return (
    <StyledHeader>
      <NavLink to="/" exact>Home</NavLink>
      <NavLink to="/favorites">Favorites</NavLink>
    </StyledHeader>
  )  
}

export default Header;
