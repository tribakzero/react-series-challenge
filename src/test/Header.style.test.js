import React from 'react';
import { shallow } from 'enzyme';
import Header from 'components/header/Header'
import 'jest-styled-components'

describe('Header', () => {
  test('renders properly', () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot()
  })
});
