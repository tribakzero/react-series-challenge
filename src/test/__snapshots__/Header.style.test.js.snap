// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          exact={true}
          to="/"
>
          Home
</NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/favorites"
>
          Favorites
</NavLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Home",
          "exact": true,
          "to": "/",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Favorites",
          "to": "/favorites",
        },
        "ref": null,
        "rendered": "Favorites",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            exact={true}
            to="/"
>
            Home
</NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/favorites"
>
            Favorites
</NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Home",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Favorites",
            "to": "/favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
